cmake/opendds_build_helpers.cmake

```diff
@@ -60,7 +60,9 @@ function(_opendds_library target)
   endif()
 
   if(MSVC AND arg_MSVC_BIGOBJ)
-    target_compile_options(${target} PRIVATE /bigobj)
+    target_compile_options(${target} PRIVATE /bigobj /W4)
+  else()
+    target_compile_options(${target} PRIVATE $<$<OR:$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleClang>,$<CXX_COMPILER_ID:GNU>>: -Wall -Wpedantic -Wno-unused -Wextra -Wcast-align -Wcast-qual -Wctor-dtor-privacy -Wdisabled-optimization -Wformat=2 -Winit-self -Wmissing-include-dirs -Woverloaded-virtual -Wredundant-decls -Wshadow -Wsign-conversion -Wstrict-overflow=5 -Wundef -Werror> $<$<CXX_COMPILER_ID:MSVC>: /W4>)
```

Comment:
A couple of thoughts about this:

1. This should be separate from bigobj because they're not related at all besides having to do with compiler options.
2. It should be broken into into separate variables because generator expressions this long are a pain to look at. I'm also not sure it needs to be a generator expression because compilers are fixed at configuration, so it could just be normal CMake code, which is (relatively) easier to read and maintain.
3. `-Werror` shouldn't be unconditional or default, it should be something we have to opt into for the CI builds. If a user in the future wants to build with a newer compiler or different settings that causes new warnings, then the build fails when it doesn't have to. GoogleTest had [this issue](https://github.com/google/googletest/issues/3447) and I think the version we use still has it.
4. This is only in `_opendds_library` and not `_opendds_executable`. They should be in common for both, probably in `_opendds_alias`.

So all together, something like:

```cmake
function(_opendds_target target) # Renamed _opendds_alias
  # This is the name the target should be exported from CMake-build OpenDDS as
  # or imported in the MPC-built OpenDDS CMake package. For consistency and
  # convenience, alias it so we can use it within the OpenDDS CMake build.
  string(REPLACE "OpenDDS_" "" name "${target}")
  set(name "OpenDDS::${name}")
  get_target_property(target_type ${target} TYPE)
  if(target_type MATCHES "LIBRARY")
    add_library("${name}" ALIAS "${target}")
  else()
    add_executable("${name}" ALIAS "${target}")
  endif()
  set_target_properties(${target} PROPERTIES EXPORT_NAME "${name}")
  
  if(OPENDDS_COMPILE_WARNINGS)
    if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
      target_compile_options(${target} PRIVATE
        -Wall -Wpedantic -Wno-unused -Wextra -Wcast-align -Wcast-qual -Wctor-dtor-privacy
        -Wdisabled-optimization -Wformat=2 -Winit-self -Wmissing-include-dirs -Woverloaded-virtual
        -Wredundant-decls -Wshadow -Wsign-conversion -Wstrict-overflow=5 -Wundef
      )
      if(OPENDDS_COMPILE_WARNINGS STREQUAL "ERROR")
        target_compile_options(${target} PRIVATE -Werror)
      endif()
    elseif(MSVC)
      target_compile_options(${target} PRIVATE /W4)
    endif()
  endif()
endfunction()
```

Then `OPENDDS_COMPILE_WARNINGS` could be default `TRUE` and you can turn off the warnings using `-DOPENDDS_COMPILE_WARNINGS=FALSE` or turn them into errors using `-DOPENDDS_COMPILE_WARNINGS=ERROR`.

`_opendds_alias` calls have to be renamed and `OPENDDS_COMPILE_WARNINGS` option has to be set in `init.cmake` and documented. We also have to set `OPENDDS_COMPILE_WARNINGS` in the workflows. Do you want me to do all that?